# SCSA: cell subtype annotation for single-cell RNA-seq data
Current most methods take manual strategies to annotate cell subtypes after clustering the single-cell RNA-seq data. Such methods are of limited application due to artificial judgment and specific marker genes of cell subtypes lacking or missing, which may lead to inefficient or false-positive identification. We present SCSA, which can automatic identify the cell subtypes quickly from single-cell RNA-seq data through  a novel developed score annotation model by using different expressed gene and cell markers reference information to improve efficiency and reduce the false-positives. Compared with other methods in real datasets, SCSA exhibits a significant performance at a high accuracy and efficient level.<br><br>
SCSA is maintained by Yinghao Cao [yhcao@ibms.pumc.edu.cn].
## Download and Installation
```
git clone https://github.com/bioinfo-ibms-pumc/SCSA.git
```
## Command Lines

```  
SCSA.py [-h] -i INPUT [-o OUTPUT] [-d DB] [-s SOURCE] [-c CLUSTER]
                 [-M MARKERDB] [-f FOLDCHANGE] [-p PVALUE] [-w WEIGHT]
                 [-g SPECIES] [-k TISSUE] [-m OUTFMT] [-T CELLTYPE]
                 [-t TARGET] [-E] [-N] [-b] [-l]

optional arguments:
  -h, --help            show this help message and exit
  -i INPUT, --input INPUT
                        Input file for marker annotation.
  -o OUTPUT, --output OUTPUT
                        Output file for marker annotation.
  -d DB, --db DB        Database for annotation. (whole.db)
  -s SOURCE, --source SOURCE
                        Source of marker genes. (cellranger,[seurat])
  -c CLUSTER, --cluster CLUSTER
                        Only deal with one cluster of marker genes.
                        (all,[1],[1,2,3],[...])
  -M MARKERDB, --MarkerDB MARKERDB
                        User-defined marker database in table format with two
                        columns.First column as Cellname, Second refers to
                        Genename.
  -f FOLDCHANGE, --foldchange FOLDCHANGE
                        Fold change threshold for marker filtering. (2.0)
  -p PVALUE, --pvalue PVALUE
                        P-value threshold for marker filtering. (0.05)
  -w WEIGHT, --weight WEIGHT
                        Weight threshold for marker filtering from cellranger
                        v1.0 results. (100)
  -g SPECIES, --species SPECIES
                        Species for annotation. Only used for cellmarker
                        database. ('Human',['Mouse'])
  -k TISSUE, --tissue TISSUE
                        Tissue for annotation. Only used for cellmarker
                        database. Run '-l' option to see all tissues.
                        ('All',[...])
  -m OUTFMT, --outfmt OUTFMT
                        Output file format for marker annotation. (ms-
                        excel,[txt])
  -T CELLTYPE, --celltype CELLTYPE
                        Cell type for annotation. (normal,[cancer])
  -t TARGET, --target TARGET
                        Target to annotation class in Database.
                        (cellmarker,[cancersea])
  -E, --Gensymbol       Using gene symbol ID instead of ensembl ID in input
                        file for calculation.
  -N, --norefdb         Only using user-defined marker database for
                        annotation.
  -b, --noprint         Do not print any detail results.
  -l, --list_tissue     List tissue names in database.
  
```
## Examples
1. To annotate a human scRNA-seq sets generated by [CellRanger](https://support.10xgenomics.com/single-cell-gene-expression/software/overview/welcome), use the following code
```
  python3 SCSA.py -d whole.db -i cellranger_pbmc_3k.csv -k All -g Human -p 0.01 -f 1.5 -m txt -o sc.txt
```
2. To annotate a human scRNA-seq sets generated by 'FindAllMarkers' function of [Seurat(Butler, A., et al. Nature Biotechnology. 2018)](https://www.nature.com/articles/nbt.4096) with ensemblIDs, use the following code
```
  python3 SCSA.py -d whole.db -s seurat -i seurat_GSE72056.csv -k All -E -g Human -p 0.01 -f 1.5
```
3. To annotate a human scRNA-seq sets generated by 'FindAllMarkers' function of [Seurat(Butler, A., et al. Nature Biotechnology. 2018)](https://www.nature.com/articles/nbt.4096) with both user-defined database and CellMarker database, use the following code
```
  python3 SCSA.py -d whole.db -i seurat_GSE72056.csv -s seurat -E -f1.5 -p 0.01 -o $i.res -m txt -M user.table 
```
4. To annotate a human scRNA-seq sets generated by [CellRanger](https://support.10xgenomics.com/single-cell-gene-expression/software/overview/welcome) only with user-defined database without any detail print, use the following code
```
  python3 SCSA.py -d whole.db -i cellranger_pbmc_3k.csv -f1.5 -p 0.01 -m txt -M user.table -N -b
```
5. To annotate cluster1 of mouse scRNA-seq sets and 
To annotate cluster1 of mouse scRNA-seq sets generated by [CellRanger](https://support.10xgenomics.com/single-cell-gene-expression/software/overview/welcome), use the following code
```
  python3 SCSA.py -d whole.db -s seurat -i seurat_mouse.csv -k All -E -g Mouse -p 0.01 -f 1 -m txt -o testout -c 1
```
6. To list tissue names in the SCSA annotation database, use the following code
```
  python3 SCSA.py -i none -d whole.db -l
```
